### PR TITLE
libclc: rename from libclc-git and update to 11.0.0

### PR DESCRIPTION
**Pull request by pullmoll**
Use the libclc src provided by llvm-project. Adopt.

### CODE REVIEW

**Comment by pullmoll** [comment]
Hmm.. cross clang11 fails. Perhaps we should use gcc instead of clang as the old template did.

**Comment by pullmoll** [comment]
Err, and again I'm searching how to undraft a PR. Where's the link?

File: srcpkgs/libclc/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+wrksrc="${pkgname}-${version}.src"
+hostmakedepends="cmake clang llvm python3 libedit-devel libffi-devel
+ ncurses-devel zlib-devel"
+short_desc="Open implementation of the OpenCL C programming language"
+maintainer="Jürgen Buchmüller <pullmoll@t-online.de>"
+license="BSD-3-Clause, MIT"
+homepage="https://libclc.llvm.org"
+distfiles="https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/libclc-${version}.src.tar.xz"
+checksum=5a0a2f67b85ede09b2f08c071c5b0b6b5ad56bfb00b39a22c369afcc613c42ee
+provides="libclc-${version}_${revision}"
```
Comment (ericonr) [inline srcpkgs/libclc/template:14]: Why is this line necessary?

Comment (pullmoll) [reply srcpkgs/libclc/template:14]: Actually I don't know if it is. There is no library, just header files, so it's probably not needed at all.

**Comment by pullmoll** [comment]
So libclc.pc is not installed for some reason. Need to fix that.

**Review by ericonr — APPROVED** [review]
Looks ok to me, but I have no way of testing :/

**Comment by pullmoll** [comment]
Hmm, and I have no idea how to test it. Nothing depends on `mesa-opencl`.
I have a Radeon RX 550 (Baffin) and believe it is not among the supported hardware,
at least there is no `usr/share/clc/baffin-*--.bc` file in `libclc-11.0.1_1.x86_64.xbps`.
Probably needs a real R600 card to test with. But then why shouldn't it work if it compiles w/o issues?

File: srcpkgs/libclc/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+checksum=5a0a2f67b85ede09b2f08c071c5b0b6b5ad56bfb00b39a22c369afcc613c42ee
+replaces="libclc-git>=0"
+
+do_configure() {
+	# We really do a native build.
+	# The outcome is just header files.
+	unset CC CXX LD CFLAGS CXXFLAGS LDFLAGS
+	mkdir -p build
+	cd build
+	cmake ${configure_args} ..
```
Comment (Vaelatern) [inline srcpkgs/libclc/template:23]: Why not the cmake build style?

Comment (pullmoll) [reply srcpkgs/libclc/template:23]: Because it tries to cross compile for the target architecture.

Comment (Vaelatern) [reply srcpkgs/libclc/template:23]: Doesn't explain why bother with the do_build or do_install steps manually

Comment (pullmoll) [reply srcpkgs/libclc/template:23]: True. I didn't think about overriding just the `do_configure()` step. I will change that.

File: srcpkgs/libclc/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+maintainer="Jürgen Buchmüller <pullmoll@t-online.de>"
+license="BSD-3-Clause, MIT"
+homepage="https://libclc.llvm.org"
+distfiles="https://github.com/llvm/llvm-project/releases/download/llvmorg-${version}/libclc-${version}.src.tar.xz"
+checksum=5a0a2f67b85ede09b2f08c071c5b0b6b5ad56bfb00b39a22c369afcc613c42ee
+replaces="libclc-git>=0"
+
+do_configure() {
+	# We really do a native build.
+	# The outcome is just header files.
```
Comment (Vaelatern) [inline srcpkgs/libclc/template:20]: Can you please clear up this comment so all can understand the purpose of this override?

```
Even when cross building we want to do a native build.
When cross building, XYZ breaks
```

Comment (pullmoll) [reply srcpkgs/libclc/template:20]: Dear, you make my life difficult.
Cross building doesn't work basically because there is no `cross-<somearch>-linux-<libc>-clang` and the `CMakeLists.txt` fails to find the compiler to use to build the output. If you want a change in that comment, please translate the following error output into a phrasing you like to see there.
```
-- The CXX compiler identification is GNU 9.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /builddir/.xbps-libclc/wrappers/aarch64-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
LLVM version: 11.0.0
LLVM system libs: 
LLVM libs: -lLLVM-11
LLVM libdir: /usr/lib
LLVM bindir: /usr/bin
LLVM ld flags: -L/usr/lib
LLVM cxx flags: -I/usr/include;-std=c++14;;;-fno-exceptions;-D_GNU_SOURCE;-D__STDC_CONSTANT_MACROS;-D__STDC_FORMAT_MACROS;-D__STDC_LIMIT_MACROS;-fno-rtti;-fno-exceptions

clang: /usr/bin/clang
llvm-as: /usr/bin/llvm-as
llvm-link: /usr/bin/llvm-link
opt: /usr/bin/opt

CMake Error: File /builddir/libclc-11.0.0.src/clc_comp_in-NOTFOUND does not exist.
CMake Error at cmake/CMakeDetermineCLCCompiler.cmake:17 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:113 (enable_language)


CMake Error: Could not find cmake module file: /builddir/libclc-11.0.0.src/build/CMakeFiles/3.18.4/CMakeCLCCompiler.cmake
CMake Error: File /builddir/libclc-11.0.0.src/ll_comp_in-NOTFOUND does not exist.
CMake Error at cmake/CMakeDetermineLLAsmCompiler.cmake:23 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:113 (enable_language)


CMake Error: Could not find cmake module file: /builddir/libclc-11.0.0.src/build/CMakeFiles/3.18.4/CMakeLLAsmCompiler.cmake
-- Check for working CLC compiler: /usr/bin/clang
CMake Error at cmake/CMakeTestCLCCompiler.cmake:24 (try_compile):
  Unknown extension ".cl" for file

    /builddir/libclc-11.0.0.src/build/CMakeFiles/CMakeTmp/testCLCCompiler.cl

  try_compile() works only for enabled languages.  Currently these are:

    CLC CXX LLAsm

  See project() command to enable other languages.
Call Stack (most recent call first):
  CMakeLists.txt:113 (enable_language)


-- Check for working CLC compiler: /usr/bin/clang -- broken
CMake Error at cmake/CMakeTestCLCCompiler.cmake:40 (message):
  The CLC compiler "/usr/bin/clang" is not able to compile a simple test
  program.

  It fails with the following output:

   

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:113 (enable_language)


-- Configuring incomplete, errors occurred!
See also "/builddir/libclc-11.0.0.src/build/CMakeFiles/CMakeOutput.log".
See also "/builddir/libclc-11.0.0.src/build/CMakeFiles/CMakeError.log".
=> ERROR: libclc-11.0.0_1: do_configure: 'CFLAGS="${CFLAGS/ -pipe / }" CXXFLAGS="${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} $(echo ${cmake_builddir}|sed 			-e 's|[^/]$|/|' -e 's|[^/]*||g' -e 's|/|../|g')' exited with 1
=> ERROR:   in do_configure() at common/build-style/cmake.sh:76
```

Comment (pullmoll) [reply srcpkgs/libclc/template:20]: Note that `libclc-git` was `archs=noarch` and this package still is in that sense, while I removed the line because we no longer want any `noarch` packages. It is a header files and `*.bc` files only package and as such using the host compiler to build it is just fine.

Comment (Vaelatern) [reply srcpkgs/libclc/template:20]: I thought that we fixed cross in cmake :( Maybe that was Meson.

I understand the purpose of the override better now, how about this as comment text, then I think this is great:

```
# Disable cross building since the cmake cross is buggy and the package
# is really just noarch, providing only text files.
```

Comment (ericonr) [reply srcpkgs/libclc/template:20]: > building since the cmake cross is buggy 

It isn't that cmake is buggy, it's more that the build requires clang explicitly, and we don't have cross clang profiles.

Comment (pullmoll) [reply srcpkgs/libclc/template:20]: How about I remove that comment? It's not that the way to configure this package will or has to change anytime sooner or later.

Comment (Vaelatern) [reply srcpkgs/libclc/template:20]: The comment is better than stumbling across it in future years and wasting time trying to fix it
